lib/active_model/serializer/adapter/json_api/link.rb

```diff
@@ -10,29 +10,40 @@ def initialize(serializer)
 
           def href(value)
             self._href = value
+            nil
           end
 
           def meta(value)
             self._meta = value
+            nil
           end
 
-          def string(value)
-            self._string = value
+          def value(value)
+            if value.respond_to?(:call)
+              string instance_eval(&value)
+              _string || to_hash
```

Comment:
I could go either way.  Like I wrote in the description 

> This commit leaves unresolved the behavior if there is both a return value to the block (as _string) and meta and/or href are set. To be decided in the receiving PR if merged.

This is more of an interface proposal than implementation.  I considered the initializer, but thought this was more similar to our other implementations, e.g. Reflection.  but I don't feel strongly about it.

I'd prefer you to merge it (if you think it's an improvement), and then add a commit or two to handle the unresolved issue I just quoted, and where the logic should go internal to the 'Link'